src/Equations/Fluid/NBIElectronTerm.cpp

```diff
@@ -0,0 +1,183 @@
+#include "DREAM/Equations/Fluid/NBIElectronTerm.hpp"
+#include "FVM/Matrix.hpp"
+#include "FVM/Grid/RadialGrid.hpp"
+#include "FVM/Grid/Grid.hpp"
+#include "FVM/UnknownQuantityHandler.hpp"
+
+
+using namespace DREAM;
+
+/**
+ * Constructor
+ */
+NBIElectronTerm::NBIElectronTerm(
+    NBIHandler* h, FVM::Grid* grid,
+    FVM::UnknownQuantityHandler* unknowns,
+    IonHandler* ionHandler
+): EquationTerm(grid), handler(h), ions(ionHandler), radialGrid(grid->GetRadialGrid()) {
+    this->unknowns = unknowns;
+    this->nr = radialGrid->GetNr();
+    this->id_ncold = unknowns->GetUnknownID(OptionConstants::UQTY_N_COLD);
+    this->id_Tcold = unknowns->GetUnknownID(OptionConstants::UQTY_T_COLD);
+    this->id_ion_density = unknowns->GetUnknownID(OptionConstants::UQTY_NI_DENS);
+    this->id_ion_temperature = unknowns->GetUnknownID(OptionConstants::UQTY_WI_ENER);
+    this->id_ni = unknowns->GetUnknownID(OptionConstants::UQTY_ION_SPECIES);
+    len_t nZ = ions->GetNZ();
+    len_t Zmax_max = 0;
+    for (len_t iz = 0; iz < nZ; ++iz)
+        Zmax_max = std::max(Zmax_max, ions->GetZ(iz));
+    len_t nCharge = Zmax_max + 1;
+    size_t deriv_size =
+        static_cast<size_t>(nr) * static_cast<size_t>(nZ) * static_cast<size_t>(nCharge);
+
+
+    Qe_1 = new real_t[nr];
+    Qe_2 = new real_t[nr];
+    Qe_3 = new real_t[nr];
+    d_Qe1_d_Te = new real_t[nr];
+    d_Qe1_d_ne = new real_t[nr];
+    d_Qe1_d_n_ij = new real_t[deriv_size];
+    d_Qe1_d_T_ij = new real_t[deriv_size];
+
+    d_Qe2_d_Te = new real_t[nr];
+    d_Qe2_d_ne = new real_t[nr];
+    d_Qe2_d_n_ij = new real_t[deriv_size];
+    d_Qe2_d_T_ij = new real_t[deriv_size];
+
+    d_Qe3_d_Te = new real_t[nr];
+    d_Qe3_d_ne = new real_t[nr];
+    d_Qe3_d_n_ij = new real_t[deriv_size];
+    d_Qe3_d_T_ij = new real_t[deriv_size];
+
+    AddUnknownForJacobian(unknowns, id_ncold);
+    AddUnknownForJacobian(unknowns, id_Tcold);
+    AddUnknownForJacobian(unknowns, id_ni);
+    AddUnknownForJacobian(unknowns, id_ion_temperature);
+}
+
+
+/**
+ * Rebuild the term (called once per time step). Call of the build function in NBIHandler
+ */
+void NBIElectronTerm::Rebuild(const real_t t, const real_t dt, FVM::UnknownQuantityHandler* unknowns) {
+    real_t beam_energy = handler->GetBeamEnergy();
+
+    // Helper to copy data
+    auto copyData = [&](real_t energy, real_t* Qe_target, real_t* d_Te_target,
+                        real_t* d_ne_target, real_t* d_n_ij_target,
+                        real_t* d_T_ij_target) {
+        handler->Build(t, dt, unknowns, energy);
```

Comment:
Should this method really be called here? This way, it will be called three times within ``NBIElectronTerm::Rebuild()``, seemingly with identical input parameters each time. For example the ``RunawayFluid`` and ``SPIHandler`` classes are built in ``src/Solver/Solver.cpp``, in the ``RebuildTerms()`` method, which is called exactly once per Newton iteration. There, you can also find infrastructure for measuring the time spent re-building the module.